micrometer-core/src/main/java/io/micrometer/core/instrument/HighCardinalityTagsDetector.java

```diff
@@ -135,44 +137,51 @@ public void close() {
 
     private void detectHighCardinalityTags() {
         try {
-            findFirst().ifPresent(this.meterNameConsumer);
+            findFirst().ifPresent((meterInfo) -> {
+                if (this.meterNameConsumer != null) {
+                    this.meterNameConsumer.accept(meterInfo.getName());
+                }
+                else {
+                    if (this.meterInfoConsumer != null) {
+                        this.meterInfoConsumer.accept(meterInfo);
+                    }
+                    else {
+                        logWarning(meterInfo);
+                    }
+                }
+            });
         }
         catch (Exception exception) {
             LOGGER.warn("Something went wrong during high cardinality tag detection", exception);
         }
     }
 
     /**
-     * Finds the name of the first Meter that potentially has high cardinality tags.
-     * @return the name of the first Meter that potentially has high cardinality tags, an
-     * empty Optional if none found.
+     * Finds the {@code HighCardinalityMeterInfo} of the first Meter that potentially has
+     * high cardinality tags.
+     * @return the {@code HighCardinalityMeterInfo} of the first Meter that potentially
+     * has high cardinality tags, an empty Optional if none found.
      */
-    public Optional<String> findFirst() {
-        Map<String, Long> meterNameFrequencies = new HashMap<>();
+    public Optional<HighCardinalityMeterInfo> findFirst() {
```

Comment:
I suppose japicmp doesn't complain because of the erased type. But effectively, I supposed this is a breaking change for any users who expected a string type when calling `get` on the returned Optional. This is unfortunate because I would like to change things as has been done here, but I'm not sure what to do about the backward incompatibility.